zipkin-receiver-kafka/src/main/scala/com/twitter/zipkin/receiver/kafka/KafkaSpanReceiver.scala

```diff
@@ -1,49 +1,55 @@
 package com.twitter.zipkin.receiver.kafka
 
-import com.twitter.app.App
 import java.util.Properties
-import com.twitter.zipkin.thriftscala.{Span => ThriftSpan}
-import com.twitter.zipkin.collector.SpanReceiver
-import com.twitter.util.{Closable, Future, Time}
+import com.twitter.app.App
 import com.twitter.finagle.stats.{DefaultStatsReceiver, StatsReceiver}
+import com.twitter.util.{Closable, Future, Time}
+import com.twitter.zipkin.collector.SpanReceiver
+import com.twitter.zipkin.thriftscala.{Span => ThriftSpan}
+import kafka.serializer.Decoder
 
 trait KafkaSpanReceiverFactory { self: App =>
-  val defaultZookeeperServer = "localhost:2181"
+  val defaultKafkaServer = "127.0.0.1:2181"
   val defaultKafkaGroupId = "zipkinId"
   val defaultKafkaZkConnectionTimeout = "1000000"
   val defaultKafkaSessionTimeout = "4000"
   val defaultKafkaSyncTime = "200"
-  val defaultKafkaAutoOffset = "largest"
+  val defaultKafkaAutoOffset = "smallest"
   val defaultKafkaTopics = Map("zipkin_kafka" -> 1)
 
   val kafkaTopics = flag[Map[String, Int]]("zipkin.kafka.topics", defaultKafkaTopics, "kafka topics to collect from")
-  val kafkaServer = flag("zipkin.kafka.server", defaultZookeeperServer, "kafka server to connect")
+  val kafkaServer = flag("zipkin.kafka.server", defaultKafkaServer, "kafka server to connect")
   val kafkaGroupId = flag("zipkin.kafka.groupid", defaultKafkaGroupId, "kafka group id")
   val kafkaZkConnectionTimeout = flag("zipkin.kafka.zk.connectionTimeout", defaultKafkaZkConnectionTimeout, "kafka zk connection timeout in ms")
   val kafkaSessionTimeout = flag("zipkin.kafka.zk.sessionTimeout", defaultKafkaSessionTimeout, "kafka zk session timeout in ms")
   val kafkaSyncTime = flag("zipkin.kafka.zk.syncTime", defaultKafkaSyncTime, "kafka zk sync time in ms")
   val kafkaAutoOffset = flag("zipkin.kafka.zk.autooffset", defaultKafkaAutoOffset, "kafka zk auto offset [smallest|largest]")
 
-  def newKafkaSpanReceiver(
+  def newKafkaSpanReceiver[T](
     process: Seq[ThriftSpan] => Future[Unit],
     stats: StatsReceiver = DefaultStatsReceiver.scope("KafkaSpanReceiver"),
-    decoder: KafkaProcessor.KafkaDecoder
+    keyDecoder: Option[Decoder[T]],
+    valueDecoder: KafkaProcessor.KafkaDecoder
   ): SpanReceiver = new SpanReceiver {
 
-    val props = new Properties() {
+    val receiverProps = new Properties() {
       put("group.id", kafkaGroupId())
       put("zookeeper.connect", kafkaServer() )
       put("zookeeper.connection.timeout.ms", kafkaZkConnectionTimeout())
       put("zookeeper.session.timeout.ms", kafkaSessionTimeout())
       put("zookeeper.sync.time.ms", kafkaSyncTime())
       put("auto.offset.reset", kafkaAutoOffset())
+      put("auto.commit.interval.ms", "10")
```

Comment:
since this is a trait, i think it makes more sense to make each of these hardcoded values into default values like above. this shouldn't dictate that you *must* make this zipkin kafka consumer id be "zipkin-consumerid"